### PR TITLE
Expand on the allocator comment in `rustc-main`

### PR DESCRIPTION
**Pull request by jyn514**
Before, it said "global_allocator does nothing". Now it gives you suggestions for what to do if you want to change the global allocator (which is likely the main reason you'd be reading the comment).

cc https://rust-lang.zulipchat.com/#narrow/stream/122651-general/topic/improving.20rustc.20memory.20usage

### CODE REVIEW

File: compiler/rustc/src/main.rs [hunk 1]
```diff
@@ -24,6 +24,15 @@
...
 // libraries. So we must reference jemalloc symbols one way or another, because
 // this file is the only object code in the rustc executable.
+//
+// NOTE: if you are reading this comment because you want to set a custom `global_allocator` for
+// benchmarking, consider using the benchmarks in the `rustc-perf` collector suite instead:
+// https://github.com/rust-lang/rustc-perf/blob/master/collector/README.md#profiling
+//
+// NOTE: if you are reading this comment because you want to replace jemalloc with another allocator
+// to compare their performance, see
+// https://github.com/fee1-dead-contrib/rust/commit/b90cfc887c31c3e7a9e6d462e2464db1fe506175#diff-43914724af6e464c1da2171e4a9b6c7e607d5bc1203fa95c0ab85be4122605ef
```
Comment (lqd) [inline compiler/rustc/src/main.rs:34]: I wonder if a link to the same commit but from our repo is more stable / less sensitive to fee1-dead changing their contrib account username or deleting their fork (or if it's all the same as far as github is concerned):

```suggestion
// https://github.com/rust-lang/rust/commit/b90cfc887c31c3e7a9e6d462e2464db1fe506175#diff-43914724af6e464c1da2171e4a9b6c7e607d5bc1203fa95c0ab85be4122605ef
```

**Comment by lqd** [comment]
r=me after that possible link improvement

**Comment by jyn514** [comment]
@bors r=lqd

thanks for the review!

**Pull request merged**